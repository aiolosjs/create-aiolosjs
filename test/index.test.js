const coffee = require('coffee');
const path = require('path');
const os = require('os');
const fs = require('fs');
const { winEOL } = require('umi-utils');

describe('test aiolosjs-create', () => {
  beforeAll(() => {
    process.env.TEST = 'test';
  });
  it('test generate aiolosjs pro project from github', async () => {
    let temp = fs.mkdtempSync(path.join(os.tmpdir(), `aiolosjs-create`));
    if (os.platform() === 'darwin') {
      temp = path.join('/private', temp);
    }
    const Coffee = coffee.Coffee;
    const fixtures = path.join(__dirname, 'fixtures');
    const response = await new Coffee({
      method: 'fork',
      cmd: path.join(__dirname, '../cli.js'),
      opt: { cwd: temp, stdout: process.stdout, stderr: process.stderr, stdin: process.stdin },
    })
      .beforeScript(path.join(fixtures, 'mock_github.js'))
      .waitForPrompt()
      .write('\n')
      .write('\n')
      .write('\n')
      .writeKey('DOWN', 'ENTER')
      .write('\n')
      .write('\n')
      .end();

    expect(winEOL(response.stdout).replace(/>/g, '❯')).toMatchSnapshot();
    expect(response.code).toBe(0);
    expect(fs.existsSync(path.join(temp, 'package.json'))).toBeTruthy();
  });
});

describe('typescript', () => {
  it('tsconfig.json should be removed if JavaScript is picked', async () => {
    let temp = fs.mkdtempSync(path.join(os.tmpdir(), `aiolosjs-create`));
    if (os.platform() === 'darwin') {
      temp = path.join('/private', temp);
    }
    const Coffee = coffee.Coffee;
    const fixtures = path.join(__dirname, 'fixtures');
    const response = await new Coffee({
      method: 'fork',
      cmd: path.join(__dirname, '../cli.js'),
      opt: { cwd: temp },
    })
      .beforeScript(path.join(fixtures, 'mock_github.js'))
      .waitForPrompt()
      .write('\n')
      .write('\n')
      .write('\n')
      .write('\n')
      .write('\n')
      .write('\n')
      .write('\n')
      .end();

    expect(response.code).toBe(0);
    expect(fs.existsSync(path.join(temp, 'jsconfig.json'))).toBeTruthy();
    expect(fs.existsSync(path.join(temp, 'tsconfig.json'))).toBeTruthy();
    expect(fs.existsSync(path.join(temp, 'package.json'))).toBeTruthy();
  });
});
