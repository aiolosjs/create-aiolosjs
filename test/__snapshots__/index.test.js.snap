// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test aiolosjs-create test generate aiolosjs pro project from github 1`] = `
"? 请选择模板类型 (Use arrow keys)
❯ ant-design-pro  - 创建一个aiolosjs pro 模板,基于antd pro. 
  app             - Create project with a simple boilerplate, support typescript
. 
  block           - Create a umi block. 
  library         - Create a library with umi. 
  plugin          - Create a umi plugin. [41D[41C[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[1A[2K[G? 请选择模板类型 ant-design-pro[31D[31C
? 🧙 AiolisJs Pro (Use arrow keys)
❯ Pro V1 [9D[9C[2K[1A[2K[G? 🧙 AiolisJs Pro Pro V1[24D[24C
? 🤓 您想使用哪种语言? (Use arrow keys)
❯ TypeScript 
  JavaScript [13D[13C[2K[1A[2K[1A[2K[G? 🤓 您想使用哪种语言? 
  TypeScript 
❯ JavaScript [13D[13C[2K[1A[2K[1A[2K[G? 🤓 您想使用哪种语言? JavaScript[33D[33C
❯ 🚚 clone success
❯ [Sylvanas] Prepare js environment...
❯ [JS] Clean up...
✨ File Generate Done
"
`;
